src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -458,30 +458,31 @@ protected override bool FindFirstChar()
 
                 return true;
             }
-            else if (_code.FCPrefix == null)
+            else if (_code.LeadingCharClasses is null)
             {
                 return true;
             }
 
-            _rightToLeft = _code.RightToLeft;
-            _caseInsensitive = _code.FCPrefix.GetValueOrDefault().CaseInsensitive;
-            string set = _code.FCPrefix.GetValueOrDefault().Prefix;
-
             // We now loop through looking for the first matching character.  This is a hot loop, so we lift out as many
             // branches as we can.  Each operation requires knowing whether this is a) right-to-left vs left-to-right, and
             // b) case-sensitive vs case-insensitive, and c) a singleton or not.  So, we split it all out into 8 loops, for
             // each combination of these. It's duplicated code, but it allows the inner loop to be much tighter than if
             // everything were combined with multiple branches on each operation.  We can also then use spans to avoid bounds
             // checks in at least the forward iteration direction where the JIT is able to detect the pattern.
 
+            // TODO https://github.com/dotnet/runtime/issues/1349:
+            // FCPrefixes may contain multiple sets, one for each of the first N characters in the expression,
+            // but the interpreter currently only uses the first set for the first character.
+
+            string set = _code.LeadingCharClasses[0].Value;
```

Comment:
`LeadingCharClasses` can never be non-null, but empty, right?

Reply:
Correct.